### PR TITLE
SONAR-6089 Allow to search Issues by Views / Sub-Views

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/IndexViewsStep.java [hunk 1]
```diff
@@ -20,22 +20,39 @@
...
   @Override
   public void execute(ComputationContext context) {
     if (context.getProject().qualifier().equals(Qualifiers.VIEW)) {
+      String viewUuid = context.getProject().uuid();
       indexer.index(context.getProject().uuid());
+
+      try {
+        ClearIndicesCacheRequest clearIndicesCacheRequest = new ClearIndicesCacheRequest();
+        // TODO why does it not work ?
+        // clearIndicesCacheRequest.filterKeys(IssueIndex.cacheKey(viewUuid));
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/IndexViewsStep.java]: dead code

Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/IndexViewsStep.java]: As you marked in TODO comment, only the relevant cache must be clear.

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java [hunk 2]
```diff
@@ -357,11 +357,16 @@ private FilterBuilder viewFilter(Collection<String> viewUuids) {
...
         .lookupType(ViewIndexDefinition.TYPE_VIEW)
         .lookupId(viewUuid)
-        .lookupPath(ViewIndexDefinition.FIELD_PROJECTS));
+        .lookupPath(ViewIndexDefinition.FIELD_PROJECTS))
+        .cacheKey(cacheKey(viewUuid));
     }
     return viewsFilter;
   }
 
+  public static String cacheKey(String viewUuid) {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java]: Method name is not appropriated. It's the cache for views lookup. I would name it "viewsLookupCacheKey()"

File: server/sonar-server/src/test/java/org/sonar/server/component/ComponentTesting.java [hunk 3]
```diff
@@ -107,7 +109,32 @@ public static ComponentDto newDeveloper(String name) {
...
 
-  private static ComponentDto newComponent(String uuid, ComponentDto module) {
+  public static ComponentDto newView(String uuid) {
+    return newProjectDto(uuid)
+      .setUuid(uuid)
+      .setScope(Scopes.PROJECT)
+      .setQualifier(Qualifiers.VIEW);
+  }
+
+  public static ComponentDto newTechnicalProject(ComponentDto project, ComponentDto view) {
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/component/ComponentTesting.java]: This DTO is about COPY_RESOURCE_ID column, so method name should be related: newProjectCopy(project, parent)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/IndexViewsStep.java (hunk 1 above)
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/IndexViewsStep.java]: Moreover this code must be moved to ViewIndexer

**Comment by julienlancelot** [comment]
The views lookup cache is now only cleared for current view, not for all views.

**Comment by simonbrandhof** [comment]
LGTM

**Pull request closed without merging**